tdigest/merging_digest.go

```diff
@@ -367,6 +382,10 @@ func (td *MergingDigest) Merge(other *MergingDigest) {
 	for i := range other.tempCentroids {
 		td.Add(other.tempCentroids[i].Mean, other.tempCentroids[i].Weight)
 	}
+
+	td.reciprocalSum = oldReciprocalSum + other.reciprocalSum
+	td.min = math.Min(other.min, td.min)
+	td.max = math.Max(other.max, td.max)
```

Comment:
these lines are unnecessary, they'll get updated by `td.Add` as the other histogram gets merged in

Reply:
That's fair, but isn't this slightly more accurate, since we only use the Mean of the smallest centroid? In practice, what kind of error does the algorithm produce?

Reply:
tdigest approximation is designed to preserve accuracy at the extreme quantiles (ie the smallest and largest samples will almost never be combined into a larger bucket), so the error between explicitly updating the min and recomputing it by adding centroids "should" be small. i can show you how to run the analysis, but i'm too lazy to do it myself so you can treat this comment as nonblocking

Reply:
Nah. I'll drop these lines for succinctness.

One case worth considering is returning a "0.001" value instead of 0. It can be potentially confusing when a value that should have a 0 min has something close to it. But most people will probably dismiss it as floating point error.